test/unit/default-transit-auth-method.bats

```diff
@@ -49,40 +49,197 @@ load _helpers
+    actual=$(echo "$object" | yq '.metadata.namespace' | tee /dev/stderr)
+     [ "${actual}" = "default" ]
+
+    actual=$(echo "$object" | yq '.spec.method' | tee /dev/stderr)
+     [ "${actual}" = "kubernetes" ]
+    actual=$(echo "$object" | yq '.spec.mount' | tee /dev/stderr)
+     [ "${actual}" = "kubernetes" ]
+    actual=$(echo "$object" | yq '.spec.kubernetes.serviceAccount' | tee /dev/stderr)
+     [ "${actual}" = "release-name-vault-secrets-operator-controller-manager" ]
+}
+
+@test "defaultTransitAuthMethod/CR: settings can be modified for kubernetes auth method" {
+    cd `chart_dir`
+    local object=$(helm template \
+        -s templates/default-transit-auth-method.yaml  \
+        --set 'controller.manager.clientCache.storageEncryption.enabled=true' \
+        --set 'controller.manager.clientCache.persistenceModel=direct-encrypted' \
+        --set 'controller.manager.clientCache.storageEncryption.namespace=tenant-2' \
+        --set 'controller.manager.clientCache.storageEncryption.mount=foo' \
+        --set 'controller.manager.clientCache.storageEncryption.kubernetes.role=role-1' \
+        --set 'controller.manager.clientCache.storageEncryption.kubernetes.serviceAccount=tenant-1' \
+        --set 'controller.manager.clientCache.storageEncryption.kubernetes.tokenAudiences={vault,foo}' \
+        --set 'controller.manager.clientCache.storageEncryption.headers=foo: bar' \
+        --set 'controller.manager.clientCache.storageEncryption.params=foo: baz' \
         . | tee /dev/stderr)
 
     local actual=$(echo "$object" | yq '.metadata.namespace' | tee /dev/stderr)
      [ "${actual}" = "default" ]
     actual=$(echo "$object" | yq '.spec.namespace' | tee /dev/stderr)
-     [ "${actual}" = "foo-ns" ]
+     [ "${actual}" = "tenant-2" ]
 
     actual=$(echo "$object" | yq '.spec.method' | tee /dev/stderr)
      [ "${actual}" = "kubernetes" ]
     actual=$(echo "$object" | yq '.spec.mount' | tee /dev/stderr)
-     [ "${actual}" = "foo-mount" ]
+     [ "${actual}" = "foo" ]
     actual=$(echo "$object" | yq '.spec.kubernetes.role' | tee /dev/stderr)
-     [ "${actual}" = "foo-role" ]
+     [ "${actual}" = "role-1" ]
     actual=$(echo "$object" | yq '.spec.kubernetes.serviceAccount' | tee /dev/stderr)
-     [ "${actual}" = "foo-sa" ]
+     [ "${actual}" = "tenant-1" ]
     actual=$(echo "$object" | yq '.spec.kubernetes.audiences' | tee /dev/stderr)
      [ "${actual}" = '["vault", "foo"]' ]
-    actual=$(echo "$object" | yq '.spec.storageEncryption.keyName' | tee /dev/stderr)
-     [ "${actual}" = "foo-keyName" ]
-    actual=$(echo "$object" | yq '.spec.storageEncryption.mount' | tee /dev/stderr)
-     [ "${actual}" = "foo-transit-mount" ]
-    actual=$(echo "$object" | yq '.spec.vaultConnectionRef' | tee /dev/stderr)
+    actual=$(echo "$object" | yq '.spec.headers.foo' | tee /dev/stderr)
+     [ "${actual}" = "bar" ]
+    actual=$(echo "$object" | yq '.spec.params.foo' | tee /dev/stderr)
```

Comment:
should we check the length of `spec.params` first?

Reply:
I'm not 100% sure it's necessary or what it would test, aside from a builtin helm function `tpl`, in the current form.
 
For each of these, if `spec.params/headers` is undefined we simply get `null` from yq and the check fails without crashing anything, here is an example:
```
@test "defaultTransitAuthMethod/CR: service account and token audiences settings can be modified for jwt auth method" {
    cd `chart_dir`
    local object=$(helm template \
        -s templates/default-transit-auth-method.yaml  \
        --set 'controller.manager.clientCache.storageEncryption.enabled=true' \
        --set 'controller.manager.clientCache.persistenceModel=direct-encrypted' \
        --set 'controller.manager.clientCache.storageEncryption.params=foo: baz' \  # <-- note no header was provided
        . | tee /dev/stderr)

     actual=$(echo "$object" | yq '.spec.headers.foo' | tee /dev/stderr)
      [ "${actual}" = "bar" ]
    actual=$(echo "$object" | yq '.spec.params.foo' | tee /dev/stderr)
      [ "${actual}" = "baz" ]
}

 ✗ defaultTransitAuthMethod/CR: test with nil h auth method
   (in test file test/unit/default-transit-auth-method.bats, line 148)
     `[ "${actual}" = "bar" ]' failed
   ---
   # Source: vault-secrets-operator/templates/default-transit-auth-method.yaml
   apiVersion: secrets.hashicorp.com/v1alpha1
   kind: VaultAuth
   metadata:
     name: release-name-vault-secrets-operator-default-transit-auth
     namespace: default
     labels:
       control-plane: controller-manager
       component: controller-manager
       app.kubernetes.io/component: controller-manager
       cacheStorageEncryption: "true"
       helm.sh/chart: vault-secrets-operator-0.1.0-beta.1
       app.kubernetes.io/name: vault-secrets-operator
       app.kubernetes.io/instance: release-name
       app.kubernetes.io/version: "0.1.0-beta.1"
       app.kubernetes.io/managed-by: Helm
   spec:
     vaultConnectionRef: default
     namespace:
     method: kubernetes
     mount: kubernetes
     kubernetes:
       role:
       serviceAccount: release-name-vault-secrets-operator-controller-manager
     params:
       foo: baz
     storageEncryption:
       keyName:
       mount:
       null  <-------- the test prints out the actual value for headers test on `spec.headers.foo` which failed here
```